compiler/src/dmd/file_manager.d

```diff
@@ -116,12 +123,57 @@ private struct PathCache
             if (!exists)
                 pathStatus.insert(st.cur, false);
             else
-                exists = pathStatus.insert(st.cur, FileName.exists(st.cur) == 2).value;
+            {
+                if (fileManager.useSourceArchive)
```

Comment:
this should be just `else if`

Reply:
See line 144. `else if` doesn't quite work here.